airflow/api_connexion/endpoints/dag_endpoint.py

```diff
@@ -180,7 +180,7 @@ def patch_dags(limit, session, offset=0, only_active=True, tags=None, dag_id_pat
     return dags_collection_schema.dump(DAGCollection(dags=dags, total_entries=total_entries))
 
 
-@security.requires_access([(permissions.ACTION_CAN_DELETE, permissions.RESOURCE_DAG)])
+@security.requires_access_dag("DELETE")
```

Comment:
Whats the difference between the current line and 
`@security.requires_access_dag("DELETE", DagAccessEntity.RUN)`
? Also should a enum/constant be used instead of the string "DELETE"? 
This applies for other methods too.

Reply:
1. The difference is just on the semantic. As of today, permissions are represented by a couple of action and resources. You can find the list of actions and resources [here](https://github.com/apache/airflow/blob/main/airflow/security/permissions.py). In AIP-56, we decided to get rid of these representations and come up with more formal/generic to represent permissions. The current list of resources is also too big and we want to make it simpler. I would recommend you reading this [comment thread](https://github.com/apache/airflow/pull/33213#discussion_r1304881823) which might help you the different decisions we took along the process.
2. We decided to go with a `Literal` instead of an `enum` (see [here](https://github.com/apache/airflow/pull/33213#discussion_r1318202392)). So yes it is a string but constraints with some specific values only